src/Paper/Paper.js

```diff
@@ -1,8 +1,8 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import classNames from 'classnames';
-import warning from 'warning';
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
```

Comment:
It seems required for React: https://github.com/react-toolbox/react-toolbox/blob/agnostic-typescript/packages/react-toolbox-core/src/components/Dialog/Dialog.tsx#L1 But why do we need this change  for the other imports?

Reply:
Most imports from non-ES2015+ modules are effectively `import namespace as`, which means the module's exports plus types (if any) - import them as a single named variable.  So `module.exports = warning` is both not a named export AND not a default export, which requires this syntax to effectively get all exports as `warning`.

Notice that our own imports with default exports need none of this, so as we eventually graduate to ES2015 modules we won't need these even with external modules, unless we choose to import all their types as well with that variable.

Effectively - this is valid spec javascript.

Reply:
Thanks for the explanation.

Reply:
Well, we have the allowSyntheticDefaultImports flag, that can make things like "import React from 'react'" valid

see: https://www.typescriptlang.org/docs/handbook/compiler-options.html
and: https://javascriptplayground.com/blog/2017/04/react-typescript/

I think that we need to be conservative and initially just change only the syntax that are really invalid

Reply:
The `allowSyntheticDefaultImports` flag doesn't actually change the semantics of the generated code, it just disables the error AFAIK, so this is not a good thing for libraries to do IMO. Webpack may compensate for it, but your tests running in node would fail. There's [a feature coming in TypeScript 2.7](https://github.com/Microsoft/TypeScript/pull/19675) which _will_ bring the semantics in line with Babel's however.

Reply:
That has also been my experience @pelotom.  Turning it on just helps find the errors quicker.